cmake/FindOpenBLAS.cmake

```diff
@@ -64,15 +81,27 @@ if(OpenBLAS_FOUND)
     message(STATUS "Found OpenBLAS libraries: ${OpenBLAS_LAPACK_LIB}")
     message(STATUS "Found OpenBLAS include: ${OpenBLAS_INCLUDE_DIR}")
   endif()
+  get_filename_component(OpenBLAS_LIB_DIR ${OpenBLAS_LIB} PATH)
+  vp_get_version_from_pkg2(blas-openblas "${OpenBLAS_LIB_DIR}/pkgconfig" OpenBLAS_VERSION)
+  if(NOT OpenBLAS_VERSION)
+    vp_get_version_from_pkg(blas-openblas "${OpenBLAS_LIB_DIR}/pkgconfig" OpenBLAS_VERSION)
```

Comment:
I have added `vp_get_version_from_pkg2` since it was for me the easiest and quickest solution to parse the OpenBLAS version without REGEX knowledge.

Looking at the summary I made for the review between the content of the different files, I see:

- Ubuntu 16.04: `openblas_config.h` and `blas-openblas.pc` (notice `blas` before) and direct `Version: 0.2.18` with capital `V` and `:`
- Ubuntu 18.04: `openblas_config.h` and `openblas.pc` with now `version=0.2.20`
- Ubuntu 20.04: `openblas_config.h` and `openblas.pc` with now `version=0.3.8`
- Ubuntu from source: `openblas_config.h` and `openblas.pc` with now `version=0.3.12.dev`

---

Looks like the cleanest way to retrieve the OpenBLAS version would be to parse the `openblas_config.h`, see in Ubuntu 16.04, 18.04, 20.04 and from source:

```
#define OPENBLAS_VERSION " OpenBLAS 0.2.18 "
#define OPENBLAS_VERSION " OpenBLAS 0.2.20 "
#define OPENBLAS_VERSION " OpenBLAS 0.3.8 "
#define OPENBLAS_VERSION " OpenBLAS 0.3.12.dev "
```